tf/tfprocess.py

```diff
@@ -1187,9 +1320,15 @@ def construct_net(self, inputs):
         else:
             h_fc5 = None
 
-        if self.moves_left:
+        # attention weights added as optional output for analysis -- ignored by backend
+        if self.POLICY_HEAD == pb.NetworkFormat.POLICY_ATTENTION:
+            if self.moves_left:
+                outputs = [h_fc1, h_fc3, h_fc5, attn_wts]
+            else:
+                outputs = [h_fc1, h_fc3, attn_wts]
+        elif self.moves_left:
```

Comment:
probably dont need these attn_wts at the end

Reply:
The attention weights are useful for analysis, and for an upcoming visualization tool. Exposing them as an output makes this easier. It shouldn't slow down training noticeably and it is not implemented in the backend.